### ISSUE TITLE
Properly Sized CSS Box for Arbitrary Map

### ISSUE THREAD

**Issue by snasphysicist** [snasphysicist/heatmap.js#13]
Following on from #12 , now that the SVG bounding box is of a correct size, the CSS box size needs to be set to properly fit arbitrary maps.

As a user of the mapping tool
I want the map SVG to fill the maximum space possible
And I want the map not to overflow out of the visible area (viewport)
So that I can see the different regions on the map as easily as possible
And so that I can see all regions on the map

**Comment by snasphysicist** [comment]
This really requires a complete CSS refactor, since it's currently a bit of a mess. My CSS is bad and I should feel bad.

**Comment by snasphysicist** [comment]
CSS refactored except for the modal stuff, which I don't currently want to touch. The only real solution I found for this was to calculate a reasonable percentage for the width of the map based on a reasonable scaling factor multiplied by the factor by which the height of the map (calculated from the SVG elements) is too large compared to the page size, then set this width from the map plotting function and allow the height to autoscale. Bit of a hack, but hands are basically tied because SVGs behave kind of weird in web pages, apparently. Added in bbef0a9, branch improvement-13.
